go/vt/tabletserver/schema_info.go

```diff
@@ -104,56 +105,53 @@ type SchemaInfo struct {
-		stats.Publish(statsPrefix+"QueryCacheLength", stats.IntFunc(si.queries.Length))
-		stats.Publish(statsPrefix+"QueryCacheSize", stats.IntFunc(si.queries.Size))
-		stats.Publish(statsPrefix+"QueryCacheCapacity", stats.IntFunc(si.queries.Capacity))
-		stats.Publish(statsPrefix+"QueryCacheOldest", stats.StringFunc(func() string {
+		queries:          cache.NewLRUCache(int64(queryCacheSize)),
+		connPool:         NewConnPool("", 3, idleTimeout, checker),
+		ticks:            timer.NewTimer(reloadTime),
+		endpoints:        endpoints,
+		reloadTime:       reloadTime,
+		queryRuleSources: NewQueryRuleInfo(),
+	}
+	schemaOnce.Do(func() {
+		stats.Publish("QueryCacheLength", stats.IntFunc(si.queries.Length))
+		stats.Publish("QueryCacheSize", stats.IntFunc(si.queries.Size))
+		stats.Publish("QueryCacheCapacity", stats.IntFunc(si.queries.Capacity))
+		stats.Publish("QueryCacheOldest", stats.StringFunc(func() string {
 			return fmt.Sprintf("%v", si.queries.Oldest())
 		}))
-		stats.Publish(statsPrefix+"SchemaReloadTime", stats.DurationFunc(si.ticks.Interval))
-		_ = stats.NewMultiCountersFunc(statsPrefix+"QueryCounts", []string{"Table", "Plan"}, si.getQueryCount)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"QueryTimesNs", []string{"Table", "Plan"}, si.getQueryTime)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"QueryRowCounts", []string{"Table", "Plan"}, si.getQueryRowCount)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"QueryErrorCounts", []string{"Table", "Plan"}, si.getQueryErrorCount)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"TableRows", []string{"Table"}, si.getTableRows)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"DataLength", []string{"Table"}, si.getDataLength)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"IndexLength", []string{"Table"}, si.getIndexLength)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"DataFree", []string{"Table"}, si.getDataFree)
-		_ = stats.NewMultiCountersFunc(statsPrefix+"MaxDataLength", []string{"Table"}, si.getMaxDataLength)
-	}
-	for _, ep := range endpoints {
-		http.Handle(ep, si)
-	}
+		stats.Publish("SchemaReloadTime", stats.DurationFunc(si.ticks.Interval))
+		_ = stats.NewMultiCountersFunc("QueryCounts", []string{"Table", "Plan"}, si.getQueryCount)
+		_ = stats.NewMultiCountersFunc("QueryTimesNs", []string{"Table", "Plan"}, si.getQueryTime)
+		_ = stats.NewMultiCountersFunc("QueryRowCounts", []string{"Table", "Plan"}, si.getQueryRowCount)
+		_ = stats.NewMultiCountersFunc("QueryErrorCounts", []string{"Table", "Plan"}, si.getQueryErrorCount)
+		_ = stats.NewMultiCountersFunc("TableRows", []string{"Table"}, si.getTableRows)
+		_ = stats.NewMultiCountersFunc("DataLength", []string{"Table"}, si.getDataLength)
+		_ = stats.NewMultiCountersFunc("IndexLength", []string{"Table"}, si.getIndexLength)
+		_ = stats.NewMultiCountersFunc("DataFree", []string{"Table"}, si.getDataFree)
+		_ = stats.NewMultiCountersFunc("MaxDataLength", []string{"Table"}, si.getMaxDataLength)
+		for _, ep := range endpoints {
```

Comment:
Please add a newline here because this part of the code is not a stats variable like the previous group of statements.